### PR TITLE
Rename orm_user to account and document as opaque-type

### LINKED ISSUE #67: User.orm_user should be renamed as User.account with an opaque type
The `orm_user` attribute of the singleuser app is an opaque object handled by the database's accounting.
It should be renamed as `account` and be given an opaque type.

### PR DESCRIPTION
**Pull request by kitchoi**
Fix #67

### CODE REVIEW

File: remoteappmanager/db/interfaces.py [hunk 1]
```diff
@@ -69,16 +69,17 @@ def get_user_by_name(self, user_name):
...
         -------
-        a User-like object that the Database understands
+        account : opaque-type
+           an object that the database understands
         """
 
     @abstractmethod
-    def get_apps_for_user(self, user):
-        """ Return an iterable of ApplicationConfig for a given User
+    def get_apps_for_user(self, account):
```
Comment (stefanoborini) [inline remoteappmanager/db/interfaces.py]: At this point I wonder if we should use get_apps_for_account instead.

File: remoteappmanager/handlers/home_handler.py [hunk 2]
```diff
@@ -90,9 +90,9 @@ def _actionhandler_start(self, options):
         _, app, policy = choice[0]
 
         container = None
-        orm_user = self.current_user.orm_user
+        user_name = self.current_user.name
         try:
-            container = yield self._start_container(orm_user,
+            container = yield self._start_container(user_name,
```
Comment (stefanoborini) [inline remoteappmanager/handlers/home_handler.py]: Why did you change this?

File: remoteappmanager/handlers/home_handler.py [hunk 3]
```diff
@@ -254,7 +252,6 @@ def _start_container(self, orm_user, app, policy, mapping_id):
         Container
         """
 
-        user_name = orm_user.name
```
Comment (stefanoborini) [inline remoteappmanager/handlers/home_handler.py]: The fact that we use only one part of an object does not imply that you should just pass that part. The API is made so that it homogeneously accepts objects. Now it's no longer so.

File: remoteappmanager/db/interfaces.py (hunk 1 above)
Comment (kitchoi) [inline remoteappmanager/db/interfaces.py]: In my previous branch I did that, but think it might be too trivial to worth breaking the API that is in a release.

File: remoteappmanager/handlers/home_handler.py (hunk 3 above)
Comment (kitchoi) [inline remoteappmanager/handlers/home_handler.py]: what the orm_user was is now an opaque type that we do not implement an interface for, so there is no guarantee and no public api for how this opaque type should behave, i.e. we can't make sure the object has an attribute `name`.  Afterall this is a private, "helper method that works with low level data".  Is the homogeneity of types of arguments to this single method worth a ABCUser?  I don't think so.

File: remoteappmanager/handlers/home_handler.py (hunk 2 above)
Comment (kitchoi) [inline remoteappmanager/handlers/home_handler.py]: We have agreed that the singleuserapp's user holds an opaque type `account`, we can no longer make assumption on what it contains.

File: remoteappmanager/handlers/home_handler.py (hunk 3 above)
Comment (stefanoborini) [inline remoteappmanager/handlers/home_handler.py]: Ok, what's the point of an opaque object if you can't basically inquire anything about its state? How do we plan to use it?

Comment (kitchoi) [inline remoteappmanager/handlers/home_handler.py]: The `account` object is only needed by the database and no need for anything else, its implementation is defined by the database (orm or csv_db) and the object's state is inquired there.

File: remoteappmanager/db/interfaces.py (hunk 1 above)
Comment (stefanoborini) [inline remoteappmanager/db/interfaces.py]: I don't think that anyone expects API consistency before 1.0...

Comment (kitchoi) [inline remoteappmanager/db/interfaces.py]: I don't mind breaking it at all if we all agree to it.

**Comment by kitchoi** [comment]
@sbo Can I merge?  I needed this for #71
